packages/editor/src/components/rich-text/index.js

```diff
@@ -910,7 +968,7 @@ export class RichText extends Component {
 									{ MultilineTag ? <MultilineTag>{ placeholder }</MultilineTag> : placeholder }
 								</Tagname>
 							}
-							{ isSelected && <FormatEdit value={ record } onChange={ this.onChange } /> }
+							{ isSelected && <FormatEdit value={ this.applyAnnotations( record, this.props.annotations ) } onChange={ this.onChange } /> }
```

Comment:
We are running `matchXPath` for *every* annotation on every render?

Reply:
It seems to me we should get rid of `matchXPath` in the `rich-text` package, and apply the path instead at `valueToFormat` and `formatToValue` in this file, where you have access to the DOM tree. There's no need to run `toTree` for every annotation.